jest-setup-after-env.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const os = require('os');
+
+const eolDescriptor = Object.getOwnPropertyDescriptor(os, 'EOL');
+
+if (!eolDescriptor) {
+	throw new TypeError('`os` must have an `EOL` property');
+}
+
+beforeAll(() => {
+	// NOTE: `jest.replaceProperty()` is unavailable for a read-only property.
+	Object.defineProperty(os, 'EOL', { ...eolDescriptor, value: '\n' });
+});
+
+afterAll(() => {
+	Object.defineProperty(os, 'EOL', eolDescriptor);
+});
```

Comment:
[ask] We may be able to move these hooks for `os.EOL` into `testRule()` like this. Any thoughts?

```js
return function testRule(schema) {
	/** @type {import('stylelint').lint} */
	let lint;
	const eolDescriptor = Object.getOwnPropertyDescriptor(os, 'EOL');

	beforeAll(() => {
		// NOTE: `jest.replaceProperty()` is unavailable for a read-only property.
		Object.defineProperty(os, 'EOL', { ...eolDescriptor, value: '\n' });

		// eslint-disable-next-line n/no-unpublished-require -- Avoid auto-install of `stylelint` peer dependency.
		lint = require('stylelint').lint;
	});

	afterAll(() => {
		Object.defineProperty(os, 'EOL', eolDescriptor);
	});
```

Reply:
I think it looks good where it is.

Reply:
Thanks. Fixed ebe4650.

Reply:
Oops. ebe4650 doesn't work expectedly 😢 

https://github.com/stylelint/stylelint/actions/runs/5666393030/job/15353040831?pr=7099#step:7:4555

Reply:
Reverted 3310063